src/pytorch_lightning/lite/wrappers.py

```diff
@@ -115,6 +118,26 @@ def _convert_float_tensor(t: Tensor) -> Tensor:
         output = apply_to_collection(output, function=_convert_float_tensor, dtype=Tensor)
         return output
 
+    @overload
+    def state_dict(self, *, destination: T_destination, prefix: str = ..., keep_vars: bool = ...) -> T_destination:
+        ...
+
+    @overload
+    def state_dict(self, *, prefix: str = ..., keep_vars: bool = ...) -> T_destination:
+        ...
+
+    def state_dict(
+        self, *args: Any, destination: Optional[T_destination] = None, prefix: str = "", keep_vars: bool = False
```

Comment:
I would remove `*args` since you aren't passing them. They are deprecated upstream anyways

```suggestion
        self, destination: Optional[T_destination] = None, prefix: str = "", keep_vars: bool = False
```